runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/SourceInputFormat.java

```diff
@@ -73,8 +69,9 @@ public void configure(Configuration configuration) {}
 
   @Override
   public void open(SourceInputSplit<T> sourceInputSplit) throws IOException {
+    options = new ObjectMapper().readValue(serializedOptions, PipelineOptions.class);
```

Comment:
Seems inefficient to decode pipeline options several times.

Is this to protect the user from mutating it?